encrypted-media/util/drm-messagehandler.js

```diff
@@ -0,0 +1,230 @@
+                return null;
+            }
+        },
+        'com.microsoft.playready': {
+            responseType: 'arraybuffer',
+            getLicenseMessage: function(response) {
+                return response;
+            },
+            getErrorResponse: function(response) {
+                return String.fromCharCode.apply(null, new Uint16Array(response));
+            },
+            getLicenseRequestFromMessage: function(message) {
+                var msg,
+                    xmlDoc;
+                var licenseRequest = null;
+                var parser = new DOMParser();
+                var dataview = new Uint16Array(message);
+
+                msg = String.fromCharCode.apply(null, dataview);
+                xmlDoc = parser.parseFromString(msg, 'application/xml');
+
+                if (xmlDoc.getElementsByTagName('Challenge')[0]) {
+                    var Challenge = xmlDoc.getElementsByTagName('Challenge')[0].childNodes[0].nodeValue;
+                    if (Challenge) {
+                        licenseRequest = atob(Challenge);
+                    }
+                }
+                return licenseRequest;
+            },
+            getRequestHeadersFromMessage: function(message) {
+                var msg,
+                    xmlDoc;
+                var headers = {};
+                var parser = new DOMParser();
+                var dataview = new Uint16Array(message);
+
+                msg = String.fromCharCode.apply(null, dataview);
+                xmlDoc = parser.parseFromString(msg, 'application/xml');
+
+                var headerNameList = xmlDoc.getElementsByTagName('name');
+                var headerValueList = xmlDoc.getElementsByTagName('value');
+                for (var i = 0; i < headerNameList.length; i++) {
+                    headers[headerNameList[i].childNodes[0].nodeValue] = headerValueList[i].childNodes[0].nodeValue;
+                }
+                // some versions of the PlayReady CDM return 'Content' instead of 'Content-Type'.
+                // this is NOT w3c conform and license servers may reject the request!
+                // -> rename it to proper w3c definition!
+                if (headers.hasOwnProperty('Content')) {
+                    headers['Content-Type'] = headers.Content;
+                    delete headers.Content;
+                }
+                return headers;
+            }
+        }
+    };
+
+    const serverTypes = {
+        'drmtoday': {
+            constructLicenseRequestUrl : function( serverURL, sessionType, messageType, content ) {
+                return serverURL;
+            },
+            getCustomHeaders : function( drmconfig, sessionType ) {
+                var customToken = { outputProtection: { digital : false, analogue: false, enforce: false },
+                                    profile: { purchase: { } },
+                                    storeLicense: ( sessionType === 'persistent-license' ) };
```

Comment:
Implementation in the W3C sense - so UA and CDM.

`storeLicense=true`, `UseSimpleNonPersistentLicense=1`, `SecureStop=1`, etc. are redundant to the behavior required by the spec (that the implementation request the appropriate license). It's possible that an implementation just ignores or certain values or otherwise not generate an appropriate request and relies on these additional parameters in the server request. Such parameters could hide implementation bugs/incompatibilities.